tools/aws.js

```diff
@@ -0,0 +1,97 @@
+
+import { execSync } from 'child_process';
+import aws4 from 'aws4';
+import fetch from 'node-fetch';
+
+/**
+ * Call AWS cli to get temporary security credentials.
+ * 
+ * Refs:
+ * - https://docs.aws.amazon.com/cli/latest/reference/sts/assume-role.html
+ * - https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_identifiers.html#identifiers-arns
+ * 
+ * @param {String} role AWS IAM ARN format
+ * @returns {Object} AWS credentials for role
+ */
+function assumeRole(role) {
+  console.log(`Assuming role ${role}`);
+  // one-liner for simplicity
+  const command = `aws sts assume-role --role-arn ${role} --duration 900 --role-session-name lambda_test`;
+  const result = execSync(command).toString();
+  return JSON.parse(result);
+}
+
+/**
+ * Fetch AWS endpoint
+ * 
+ * @param {Object} options 
+ * @param {Object} credentials 
+ * @returns {Promise}
+ */
+async function fetchAWS(options = {}, credentials = {}) {
+  const awsURL = new URL(options.url);
+  console.log('Calling psoxy...');
+  console.log(`Request: ${options.url}`);
+  
+  // aws4 is not able to parse a full URL:
+  // https://[id].lambda-url.us-east-1.on.aws/
+  // the result is missing `service` and `region` which are mandatory
+  const signed = aws4.sign({
+    host: awsURL.host,
+    path: awsURL.pathname,
+    service: 'lambda',
+    region: awsURL.host.split('.')[2],
+  }, {
+    accessKeyId: credentials?.Credentials.AccessKeyId,
+    secretAccessKey: credentials?.Credentials.SecretAccessKey,
+    sessionToken: credentials?.Credentials.SessionToken,
+  });
+
+  const headers = {
+    ...signed.headers,
+    '-Haccept-encoding': options.gzip ? 'gzip' : 'none',
+    '-HX-Psoxy-Skip-Sanitizer': options.skip.toString()
+  }
+  console.log('Waiting response...');
+
+  if (options.verbose) {
+    console.log('Options:');
+    console.log(options);
+    console.log('Request headers:');
+    console.log(headers);
+  }
+
+  return await fetch(options.url, {headers: headers});
```

Comment:
Not for this PR, but this will need to support POST operations with body; for example for Dropbox as that API works in that way -only POST, no GETs-